p256/src/arithmetic/affine.rs

```diff
@@ -1,500 +0,0 @@
-    type Output = ProjectivePoint;
-
-    fn mul(self, scalar: &Scalar) -> ProjectivePoint {
-        ProjectivePoint::from(self) * scalar
-    }
-}
-
-impl Neg for AffinePoint {
-    type Output = AffinePoint;
-
-    fn neg(self) -> Self::Output {
-        AffinePoint {
-            x: self.x,
-            y: -self.y,
-            infinity: self.infinity,
-        }
-    }
-}
-
-impl DecompressPoint<NistP256> for AffinePoint {
-    fn decompress(x_bytes: &FieldBytes, y_is_odd: Choice) -> CtOption<Self> {
-        FieldElement::from_bytes(x_bytes).and_then(|x| {
-            let alpha = x * &x * &x + &(CURVE_EQUATION_A * &x) + &CURVE_EQUATION_B;
-            let beta = alpha.sqrt();
-
-            beta.map(|beta| {
-                let y = FieldElement::conditional_select(
-                    &(MODULUS - &beta),
-                    &beta,
-                    beta.is_odd().ct_eq(&y_is_odd),
-                );
-
-                Self { x, y, infinity: 0 }
-            })
-        })
-    }
-}
-
-impl GroupEncoding for AffinePoint {
-    type Repr = CompressedPoint;
-
-    /// NOTE: not constant-time with respect to identity point
-    fn from_bytes(bytes: &Self::Repr) -> CtOption<Self> {
-        EncodedPoint::from_bytes(bytes)
-            .map(|point| CtOption::new(point, Choice::from(1)))
-            .unwrap_or_else(|_| {
-                // SEC1 identity encoding is technically 1-byte 0x00, but the
-                // `GroupEncoding` API requires a fixed-width `Repr`
-                let is_identity = bytes.ct_eq(&Self::Repr::default());
-                CtOption::new(EncodedPoint::identity(), is_identity)
-            })
-            .and_then(|point| Self::from_encoded_point(&point))
-    }
-
-    fn from_bytes_unchecked(bytes: &Self::Repr) -> CtOption<Self> {
-        // No unchecked conversion possible for compressed points
-        Self::from_bytes(bytes)
-    }
-
-    fn to_bytes(&self) -> Self::Repr {
-        let encoded = self.to_encoded_point(true);
-        let mut result = CompressedPoint::default();
-        result[..encoded.len()].copy_from_slice(encoded.as_bytes());
-        result
-    }
-}
-
-impl DecompactPoint<NistP256> for AffinePoint {
-    fn decompact(x_bytes: &FieldBytes) -> CtOption<Self> {
-        FieldElement::from_bytes(x_bytes).and_then(|x| {
-            let montgomery_y = (x * &x * &x + &(CURVE_EQUATION_A * &x) + &CURVE_EQUATION_B).sqrt();
-            montgomery_y.map(|montgomery_y| {
-                // Convert to canonical form for comparisons
-                let y = montgomery_y.to_canonical();
-                let p_y = MODULUS.subtract(&y);
-                let (_, borrow) = p_y.informed_subtract(&y);
-                let recovered_y = if borrow == 0 { y } else { p_y };
-                AffinePoint {
-                    x,
-                    y: recovered_y.to_montgomery(),
-                    infinity: 0,
-                }
-            })
-        })
-    }
-}
```

Comment:
One regression: I didn't port this across yet. It seems like it could be impl'd generically, however